scopes/ui-foundation/ui/start-cmd/start.cmd.tsx

```diff
@@ -46,7 +47,7 @@ export class StartCmd implements Command {
     return report([uiRootName, userPattern], { dev, port, rebuild }, this.ui, this.logger, this.pubsub);
   }
 
-  private asyncRender(startingTimestamp, pubsub, commandFlags, uiServer) {
+  private asyncRender(startingTimestamp: number, pubsub: PubsubMain, commandFlags: Object, uiServer: UIServer) {
```

Comment:
Please add types. `Object` is not really useful. I have no idea what flags to expect when visiting this function.

Reply:
@davidfirst 
This is exactly my point in our conversation. 
Now every time we want to add/remove/change a new flag in a command, we need to look everywhere for each type that hardcoded the flag or part of them in the type definition and refactor it.  That will significantly slow down the development

If you want to see witch flags exist, you should look at the right place where they are defined instead of defining them in multiple places and creating two (or more) sources of truth.

Is that correct?

Reply:
Not exactly.
You never need to define your types multiple times, you can re-use them.
Something like `export type StartFlags = { verbose: boolean, .... }`.
Then, everywhere else, just use StartFlags. When you add a new flag, you only add it to this `StartFlags` type, one place.